pvlib/spectrum/mismatch.py

```diff
@@ -1100,3 +1101,20 @@ def qe_to_sr(qe, wavelength=None, normalize=False):
         spectral_responsivity = normalize_max2one(spectral_responsivity)
 
     return spectral_responsivity
+
+
+def average_photon_energy(spectral_irr):
+    si = spectral_irr
```

Comment:
`spectral_irr` is a pandas object; consequently, it is mutable. Be careful when assigning it to a new variable. Fortunately, the methods applied to `spectral_irr` in this function produce new objects. Maybe this line isn't even necessary.

Reply:
Thanks for picking up on this. My thinking behind this line was just to have the argument as something easily understood by the user (`spectral_irr` is clearly spectral irradiance) but then have something simpler and shorter in the code that the user doesn't see--- `si`. You raise a fair point though.

> the methods applied to spectral_irr in this function produce new objects

So, in this case, do you think it is okay to leave it as it is?

Reply:
Thank you for raising the issue [#2150 ](https://github.com/pvlib/pvlib-python/issues/2150). This should get a solution there.